core/src/main/java/org/jdbi/v3/ColonPrefixNamedParamStatementRewriter.java

```diff
@@ -150,18 +153,18 @@ public void bind(Binding params, PreparedStatement statement) throws SQLExceptio
                     if ("*".equals(named_param)) {
                         continue;
                     }
-                    Argument a = params.forName(named_param);
-                    if (a == null) {
-                        a = params.forPosition(i);
-                    }
-
-                    if (a == null) {
-                        String msg = String.format("Unable to execute, no named parameter matches " +
-                                                   "\"%s\" and no positional param for place %d (which is %d in " +
-                                                   "the JDBC 'start at 1' scheme) has been set.",
-                                                   named_param, i, i + 1);
-                        throw new UnableToExecuteStatementException(msg, context);
-                    }
+                    final int index = i;
+                    Argument a = Stream.concat(
+                            JdbiStreams.toStream(params.findForName(named_param)),
+                            JdbiStreams.toStream(params.findForPosition(i)))
```

Comment:
I found a lazily-evaluated solution for this on stackoverflow. It's off topic but I'll push it to this branch since it's not worth its own PR.